src/jsd/platinum_actuator.cc

```diff
@@ -171,128 +192,186 @@ bool fastcat::PlatinumActuator::HandleNewProfTorqueCmdImpl(const DeviceCmd& cmd)
 {
   if (IsMotionFaultConditionMet()) {
-    ERROR("Act %s: %s", name_.c_str(), "Fault condition present, faulting");
+    ERROR("Act %s: %s", name_.c_str(), "Fault Condition present, faulting");
     return ALL_DEVICE_FAULT;
   }
 
-  // Transition to ACTUATOR_SMS_HOLDING once profile execution is complete if
-  // the option to actively hold position is not on.
-  if (state_->platinum_actuator_state.target_reached &&
-      !params_.prof_pos_hold) {
-    TransitionToState(ACTUATOR_SMS_HOLDING);
+  if (state_->platinum_actuator_state.servo_enabled) {
+    // If brakes are disengaged, setup the traps and transition to the execution
+    // state
+    fastcat_trap_generate_vel(
+        &trap_, state_->time, 0, 0,
+        last_cmd_.actuator_prof_torque_cmd.target_torque_amps,
+        params_.torque_slope_amps_per_sec,
+        last_cmd_.actuator_prof_torque_cmd.max_duration);
+
+    TransitionToState(ACTUATOR_SMS_PROF_TORQUE);
+
+  } else {
+    // Otherwise, command the current position to trigger the transition and
+    // wait
+    jsd_elmo_motion_command_cst_t jsd_cmd;
+
+    jsd_cmd.target_torque_amps = 0;
+    jsd_cmd.torque_offset_amps = 0;
+
+    ElmoCST(jsd_cmd);
+
+    if ((state_->monotonic_time - last_transition_time_) >
+        (prof_disengaging_timeout_ + 2 * loop_period_)) {
+      ERROR("Act %s: Brake Disengage %lf sec timeout expired, faulting",
+            name_.c_str(), prof_disengaging_timeout_);
+
+      fastcat_fault_ = ACTUATOR_FASTCAT_FAULT_BRAKE_DISENGAGE_TIMEOUT_EXCEEDED;
+      return ALL_DEVICE_FAULT;
+    }
   }
 
   return NO_FAULT;
 }
 
+fastcat::FaultType fastcat::PlatinumActuator::ProcessProfPos()
+{
+  return ProcessProfPosTrapImpl();
+}
+
 fastcat::FaultType fastcat::PlatinumActuator::ProcessProfVel()
 {
   if (IsMotionFaultConditionMet()) {
-    ERROR("Act %s: %s", name_.c_str(), "Fault condition present, faulting");
+    ERROR("Act %s: %s", name_.c_str(), "Fault Condition present, faulting");
     return ALL_DEVICE_FAULT;
   }
 
-  jsd_elmo_motion_command_prof_vel_t jsd_cmd;
-  // If max_duration is greater than zero, zero out the target after the
-  // commanded duration.
-  if (last_cmd_.actuator_prof_vel_cmd.max_duration > 1e-9 &&
-      (state_->monotonic_time - last_transition_time_) >
-          last_cmd_.actuator_prof_vel_cmd.max_duration) {
-    TransitionToState(ACTUATOR_SMS_HOLDING);
+  jsd_elmo_motion_command_csv_t jsd_cmd;
 
-    jsd_cmd.target_velocity = 0;
-    jsd_cmd.profile_accel =
-        EuToCnts(last_cmd_.actuator_prof_vel_cmd.profile_accel);
-    jsd_cmd.profile_decel =
-        EuToCnts(last_cmd_.actuator_prof_vel_cmd.profile_accel);
-  } else {
-    jsd_cmd.target_velocity =
-        EuToCnts(last_cmd_.actuator_prof_vel_cmd.target_velocity);
-    jsd_cmd.profile_accel =
-        EuToCnts(last_cmd_.actuator_prof_vel_cmd.profile_accel);
-    jsd_cmd.profile_decel =
-        EuToCnts(last_cmd_.actuator_prof_vel_cmd.profile_accel);
-  }
+  double pos_eu, vel;
```

Comment:
Nit, initialize variables to some default value. Don't rely on fastcat_trap_update_vel to initialize the values for you
```
double pos_eu = 0.0;
double vel = 0.0;
```

Reply:
@kwehage This is how it was done in multiple locations previously. I agree with this nit and have changed it in multiple locations :)